src/js_native_api_types.h

```diff
@@ -115,4 +115,25 @@ typedef struct {
   napi_status error_code;
 } napi_extended_error_info;
 
+#ifdef NAPI_EXPERIMENTAL
+typedef enum {
+  napi_key_include_prototypes,
+  napi_key_own_only
+} napi_key_collection_mode;
+
+typedef enum {
+  napi_key_all_properties = 0,
+  napi_key_writable = 1,
+  napi_key_enumerable = 1 << 1,
+  napi_key_configurable = 1 << 2,
+  napi_key_skip_strings = 1 << 3,
+  napi_key_skip_symbols = 1 << 4
+} napi_key_filter;
+
+typedef enum {
+  napi_key_keep_numbers,
+  napi_key_numbers_to_strings
+} napi_key_conversion;
+#endif
```

Comment:
Would be great if this #endif had a comment to tie it to #ifdef NAPI_EXPERIMENTAL, but just a nit.